Project/Source/Components/ComponentAnimation.cpp

```diff
@@ -68,26 +90,18 @@ void ComponentAnimation::OnEditorUpdate() {
 	UID oldStateMachinePrincipalUID = stateMachineResourceUIDPrincipal;
 	ImGui::ResourceSlot<ResourceStateMachine>("State Machine Principal", &stateMachineResourceUIDPrincipal);
 	if (oldStateMachinePrincipalUID != stateMachineResourceUIDPrincipal) {
-		ResourceStateMachine* resourceStateMachine = App->resources->GetResource<ResourceStateMachine>(stateMachineResourceUIDPrincipal);
-		if (resourceStateMachine) {
-			currentStatePrincipal = resourceStateMachine->initialState;
-			loadedResourceStateMachine = true;
-		} else {
-			loadedResourceStateMachine = false;
-		}
+		currentStatePrincipal = State();
+		currentTimeStatesPrincipal.clear();
+		loadedResourceStateMachine = false;
```

Comment:
loadedResourceStateMachine = false;
Will it be set to false if a state machine is loaded, and then the user changes the state machine reference by drag&dropping.
Or the drag&drop will call LoadStateMachines somehow and set it to true?